src/python/pants/backend/python/rules/download_pex_bin.py

```diff
@@ -62,31 +62,42 @@ def create_execute_request(  # type: ignore[override]
         pex_args: Iterable[str],
         description: str,
         input_files: Optional[Digest] = None,
+        env: Optional[Dict[str, str]] = None,
```

Comment:
Better to use `typing.Mapping` so that this works with e.g. `FrozenDict` and `OrderedDict`.

Reply:
I only fixed the two local cases. This spirals outward quite aways.

Reply:
Thank you!

> This spirals outward quite aways.

That was my fear.